server/utils/assessmentUtils.ts

```diff
@@ -25,35 +25,49 @@ export const formatAssessmentData = (
     .filter(areaConfig => areaConfig.crimNeedsKey in crimNeeds && crimNeeds[areaConfig.crimNeedsKey])
     .map(areaConfig => {
       let score
-      const isLinkedtoRoSH = crimNeeds[areaConfig.crimNeedsKey][`${areaConfig.crimNeedsSubKey}LinkedToHarm`] === 'YES'
-      const isLinkedtoReoffending =
-        crimNeeds[areaConfig.crimNeedsKey][`${areaConfig.crimNeedsSubKey}LinkedToReoffending`] === 'YES'
-      const isLinkedtoStrengthsOrProtectiveFactors =
-        crimNeeds[areaConfig.crimNeedsKey][`${areaConfig.crimNeedsSubKey}Strengths`] === 'YES'
+
+      const linkedtoHarm = crimNeeds[areaConfig.crimNeedsKey][`${areaConfig.crimNeedsSubKey}LinkedToHarm`].toLowerCase()
+      const linkedtoReoffending =
+        crimNeeds[areaConfig.crimNeedsKey][`${areaConfig.crimNeedsSubKey}LinkedToReoffending`].toLowerCase()
+      const linkedtoStrengthsOrProtectiveFactors =
+        crimNeeds[areaConfig.crimNeedsKey][`${areaConfig.crimNeedsSubKey}Strengths`].toLowerCase()
+
       let subData: SubAreaData
       let overallScore
       const motivationToMakeChanges = motivationText(
         assessment.sanAssessmentData[`${areaConfig.assessmentKey}_changes`]?.value,
       )
 
-      const linkedToRoSHYesNo = isLinkedtoRoSH ? 'yes' : 'no'
-      const linkedToReoffendingYesNo = isLinkedtoReoffending ? 'yes' : 'no'
-      const linkedtoStrengthsOrProtectiveFactorsYesNo = isLinkedtoStrengthsOrProtectiveFactors ? 'yes' : 'no'
-
       const riskOfSeriousHarmDetails =
-        assessment.sanAssessmentData[
-          `${areaConfig.assessmentKey}_practitioner_analysis_risk_of_serious_harm_${linkedToRoSHYesNo}_details`
-        ]?.value
+        linkedtoHarm !== 'null'
```

Comment:
is 'null' a possible value here? If the field is missing in line 29, what does calling "toLowerCase" on it do? I'm guessing it throws an exception so we won't get to this line?

Reply:
Discussed this on slack, adding here for visibility. The value is sent to us as "NULL" rather than being undefined, so it is a string. We do .toLowerCase() so we can call the correct key for each field such as ..._practitioner_analysis_risk_of_serious_harm_**yes**_details. Added comment in the code to make this clearer in the future